doc/DEPENDENCIES.md

```diff
@@ -8,8 +8,10 @@
 | [C/C++ Compiler](#c-and-c-compiler) | external | see below |  |
 | [CMake](#cmake) | external | \>= `3.12` | |
 | [Boost (Headers Only)](#boost-headers-only) | external | `1.72.0` | headers only library |
-| [Python 3 Libraries](#python-3-libraries) | external | \> `3.6.12` | Can be [excluded](#overriding-python-dependency). Requires ``numpy`` package |
+| [MPI](https://www.mpi-forum.org) | external | No current implementation or version requirements | Required for [multi-process distributed execution](DISTRIBUTED_PROCESSING.md) |
+| [Python 3 Libraries](#python-3-libraries) | external | \> `3.6.8` | Can be [excluded](#overriding-python-dependency). Requires ``numpy`` package |
```

Comment:
I was recently reminded that python [3.6 EOL](https://endoflife.date/python) is in December.  We might consider making 3.7 our minimum.